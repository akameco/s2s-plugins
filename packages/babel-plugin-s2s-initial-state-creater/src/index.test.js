// @flow
import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from '.'

const testWithFixture = (title, fixture) => {
  pluginTester({
    plugin,
    snapshot: true,
    pluginOptions: { from: fixture },
    tests: [
      {
        title,
        fixture,
      },
    ],
  })
}

const getFixturePath = (...x) =>
  path.resolve(__dirname, '..', '__fixtures__', ...x)

testWithFixture('initialStateがある場合', getFixturePath('reducer.js'))
testWithFixture('initialStateがない場合', getFixturePath('reducer-no-state.js'))
