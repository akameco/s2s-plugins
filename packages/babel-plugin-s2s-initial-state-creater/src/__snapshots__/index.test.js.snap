// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initialStateがある場合 1`] = `
"
// @flow
type Count = number
type State = { count: Count, isLoading: boolean }

// $FlowFixMe
const initialState: State = {}

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
type Count = number;
type State = { count: Count; isLoading: boolean; };

// $FlowFixMe
const initialState: State = { \\"count\\": 0, \\"isLoading\\": false };
"
`;

exports[`initialStateがない場合 1`] = `
"
// @flow
type Count = number
type State = { count: Count, isLoading: boolean }

const initState = {}

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
type Count = number;
type State = { count: Count; isLoading: boolean; };

const initState = {};
"
`;
